src/store/messages/saga.ts

```diff
@@ -90,6 +92,45 @@ const _isActive = (channelId) => (state) => {
 
 const FETCH_CHAT_CHANNEL_INTERVAL = 60000;
 
+function* getZeroUsersMap() {
+  const users = yield select((state) => state.normalized.users);
+  const zeroUsersMap: { [matrixId: string]: User } = {};
+  for (const user of Object.values(users)) {
+    zeroUsersMap[(user as User).matrixId] = user as User;
+  }
+  // map current user as well
+  const currentUser = yield select(currentUserSelector());
+  zeroUsersMap[currentUser.matrixId] = {
+    userId: currentUser.id,
+    profileId: currentUser.profileSummary.id,
+    firstName: currentUser.profileSummary.firstName,
+    lastName: currentUser.profileSummary.lastName,
+    profileImage: currentUser.profileSummary.profileImage,
+  } as User;
+
+  return zeroUsersMap;
+}
+
+export function* mapMessageSenders(messages) {
+  if (!featureFlags.enableMatrix) {
+    return;
+  }
+
+  const zeroUsersMap = yield call(getZeroUsersMap);
+  messages.forEach((message) => {
+    const zeroUser = zeroUsersMap[message.sender.userId];
+    if (zeroUser) {
```

Comment:
The assumption here is that we already know about the user. Is it possible that we've never seen this user before?

Reply:
In most cases we would have the user. The logic being, we first load all the channles + conversations, which has all the "otherMembers" in a channel, which gets normalized and saved in the user state (in redux). After which we fetch the messages in channel, so assuming that a message was sent by someone who is in a channel would be in the `otherMembers` field, which would be cached.

The only case there will be a miss would be when the user has left the channel. I am looking into that currently, since it's required in displaying the message sender's name in the "parent message" as well (related to replies). 

If there's any other case which i have missed then please let me know